### DISCUSSION TITLE
[@types/oidc-provider] Default export not working with ESM

### DISCUSSION THREAD

**Discussion by dinassh** [DefinitelyTyped/DefinitelyTyped#64148] [Issues with a @types package] [answered]
The module's default export type is unresolved when importing in ESM project. I'm using `@types/oidc-provider` version `8.1.0`, TS version `4.9.4`, compiler options 
```json
//tsconfig.json
"compilerOptions": {
     "target": "ESNext",
     "module": "NodeNext",
     "moduleResolution": "NodeNext",
     "esModuleInterop": true,
     "strict": true,
}
```

When importing the module's default export like
```ts
import Provider from 'oidc-provider'
```
the type isn't properly resolved:
```
type Provider = /*unresolved*/ any
Cannot use namespace 'Provider' as a type.
```
Trying to use it results in the TS error:
```ts
const provider = new Provider(/*...*/)
```
```
This expression is not constructable.
  Type 'typeof import("node_modules/@types/oidc-provider/index")' has no construct signatures.
```

Looks like the `index.d.ts` file is interpreted by the TS as a CJS module, similarly to this issue https://github.com/microsoft/TypeScript/issues/50466

Patching the `package.json` file to have `"type": "module"` helps.

[a comment by an automated account is not shown]

**Comment by BendingBender** [comment, marked as answer]
See https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64331
